ruby/workflows/account_transfer_workflow_scenarios.rb

```diff
@@ -0,0 +1,123 @@
+# frozen_string_literal: true
+
+require_relative 'account_transfer_workflow'
+require 'temporalio/error'
+require 'timeout'
+
+module Workflows
+  class AccountTransferWorkflowScenarios < AccountTransferWorkflow
+    # This workflow is designed to demonstrate various scenarios for the Account Transfer Workflow.
+
+    # `workflow_dynamic` makes this workflow to catch all the workflow types that is not otherwise
+    # registered by the worker, i.e. AccountTransferWorkflowRecoverableFailure,
+    # AccountTransferWorkflowHumanInLoop and AccountTransferWorkflowAdvancedVisibility.
+    workflow_dynamic
+
+    BUG = 'AccountTransferWorkflowRecoverableFailure'
+    NEEDS_APPROVAL = 'AccountTransferWorkflowHumanInLoop'
+    ADVANCED_VISIBILITY = 'AccountTransferWorkflowAdvancedVisibility'
```

Comment:
This is a bit confusing for readers of this repository. If this is some kind of test-only utility, it should be moved there. If there is a legitimate need for these 3 types of workflows, would recommend explicit workflows for them instead of leaning on dynamic workflows.